### ISSUE TITLE
No longer works on macOS 15 (Sequoia)

### ISSUE THREAD

**Issue by johnm1403** [martinsoft/magic-battery-checker#1]
Apple have changed the way devices are reported by `ioreg` on Sequoia. Now looks like the devices are reported as `AppleMultitouchDevice` but haven't seen a way to get the battery level :/

**Comment by johnm1403** [comment]
Fixed. `btstat` has been rewritten to parse output from `ioreg` differently. 
`btstat` will now output device id, device type, serial number, and battery percentage. Adding either device id or serial number as a parameter will output just the battery level for that device
e.g.
```
$ btstat
0x1122334455 "Magic Mouse" AB12DE45ED67BA 35
0x2233445566 "Magic Keyboard" CD12EF45ED67BA 21

$ btstat AB12DE45ED67BA
35
```

**Comment by ahu** [comment]
Today I upgrade my macOS to 15 and also my btstat file. I got nothing after I run it.

I then run `ioreg -c AppleDeviceManagementHIDEventService -r -l` and got these below:

```
+-o AppleDeviceManagementHIDEventService  <class AppleDeviceManagementHIDEventS$
    {
      "IOGeneralInterest" = "IOCommand is not serializable"
      "IOMatchedAtBoot" = Yes
      "LowBatteryNotificationPercentage" = 6
      "PrimaryUsagePage" = 65280
      "BatteryFaultNotificationType" = "MOBatteryFault"
      "HasBattery" = Yes
      "VendorID" = 76
      "VersionNumber" = 785
      "Built-In" = No
      "DeviceAddress" = "dc-a9-04-01-1f-46"
      "BatteryLowPowerDisconnectionNotificationType" = "MOLowBatteryPowerDiscon$
      "CurrentBatteryPercentNotificationType" = "CurrentBatteryPercent"
      "Product" =
      "SerialNumber" = "DC:A9:04:01:1F:46"
      "WakeReason" = "Button (0x03)"
      "Transport" = "Bluetooth"
      "BatteryLowNotificationType" = "MOLowBattery"
      "Manufacturer" = "Apple Inc."
      "ConnectionNotificationType" = "MOConnected"
      "ProductID" = 617
      "DeviceUsagePairs" = ({"DeviceUsagePage"=65280,"DeviceUsage"=11},{"Device$
      "IOPersonalityPublisher" = "com.apple.driver.AppleTopCaseDriverV2"
      "PoweredOffNotificationType" = "MOOff"
      "MTFW Version" = 1301
      "BD_ADDR" = <dca904011f46>
      "BatteryPercent" = 84
      "HostTransportSwitchNotificationType" = "MOTransportSwitch"
      "CriticallyLowBatteryNotificationPercentage" = 3
      "ReportInterval" = 8000
      "RadioFW Version" = 785
      "VendorIDSource" = 1
      "STFW Version" = 2164
      "CFBundleIdentifier" = "com.apple.driver.AppleTopCaseHIDEventDriver"
      "IOProviderClass" = "IOHIDInterface"
      "LocationID" = 67182406
      "BluetoothDevice" = Yes
      "IOClass" = "AppleDeviceManagementHIDEventService"
      "DisconnectionNotificationType" = "MODisconnected"
      "HIDServiceSupport" = No
      "ProductIDArray" = (617)
      "CFBundleIdentifierKernel" = "com.apple.driver.AppleTopCaseHIDEventDriver$
      "BatteryStatusFlags" = 0
      "IOMatchCategory" = "IODefaultMatchCategory"
      "ColorID" = 0
      "IOProbeScore" = 7175
      "CountryCode" = 0
      "PrimaryUsage" = 11
      "BTFW Version" = 785
    }


+-o AppleDeviceManagementHIDEventService  <class AppleDeviceManagementHIDEventS$
    {
      "IOGeneralInterest" = "IOCommand is not serializable"
      "IOMatchedAtBoot" = Yes
      "LowBatteryNotificationPercentage" = 6
      "PrimaryUsagePage" = 65280
      "BatteryFaultNotificationType" = "KBBatteryFault"
      "HasBattery" = Yes
      "VendorID" = 76
      "VersionNumber" = 518
      "Built-In" = No
      "DeviceAddress" = "70-f0-87-35-1f-92"
      "BatteryLowPowerDisconnectionNotificationType" = "KBLowBatteryPowerDiscon$
      "WakeReason" = "Keyboard (0x02)"
      "Product" =
      "SerialNumber" = "70:F0:87:35:1F:92"
      "Transport" = "Bluetooth"
      "BatteryLowNotificationType" = "KBLowBattery"
      "Manufacturer" = "Apple Inc."
      "ConnectionNotificationType" = "KBConnected"
      "ProductID" = 615
      "DeviceUsagePairs" = ({"DeviceUsagePage"=65280,"DeviceUsage"=11},{"Device$
      "IOPersonalityPublisher" = "com.apple.driver.AppleTopCaseHIDEventDriver"
      "PoweredOffNotificationType" = "KBOff"
      "BD_ADDR" = <70f087351f92>
      "BatteryPercent" = 95
      "HostTransportSwitchNotificationType" = "KBTransportSwitch"
      "CriticallyLowBatteryNotificationPercentage" = 3
      "ReportInterval" = 8000
      "RadioFW Version" = 518
      "VendorIDSource" = 1
      "STFW Version" = 2160
      "CFBundleIdentifier" = "com.apple.driver.AppleTopCaseHIDEventDriver"
      "IOProviderClass" = "IOHIDInterface"
      "LocationID" = 120922002
      "BluetoothDevice" = Yes
      "IOClass" = "AppleDeviceManagementHIDEventService"
      "DisconnectionNotificationType" = "KBDisconnected"
      "HIDServiceSupport" = No
      "ProductIDArray" = (615)
      "CFBundleIdentifierKernel" = "com.apple.driver.AppleTopCaseHIDEventDriver$
      "BatteryStatusFlags" = 0
      "IOMatchCategory" = "IODefaultMatchCategory"
      "ColorID" = 0
      "IOProbeScore" = 7175
      "CountryCode" = 0
      "PrimaryUsage" = 11
      "BTFW Version" = 518
    }
```

Please help me how could I fix it.
